azext_iot/digitaltwins/providers/model.py

```diff
@@ -76,10 +88,61 @@ def add(self, models=None, from_directory=None):
 
         logger.info("Models payload %s", json.dumps(payload))
 
-        # @vilit - hack to customize 403's to have more specific error messages
+        models_created = []
         try:
+            # Process models in batches if models to process exceed the API limit
+            if len(payload) > MAX_MODELS_API_LIMIT:
+
+                model_id_to_model_map = {}
+                for model_def in payload:
+                    model_id_to_model_map[model_def['@id']] = model_def
+
+                # Create a dictionary to categorize models by their number of dependencies
+                dep_count_to_models_map = {}
+                for model in payload:
+                    num_dependencies = len(get_model_dependencies(model, model_id_to_model_map))
+                    if num_dependencies not in dep_count_to_models_map:
+                        dep_count_to_models_map[num_dependencies] = []
+                    dep_count_to_models_map[num_dependencies].append(model)
+
+                # Sort the dictionary by dependency count
+                dep_count_to_models_map = dict(sorted(dep_count_to_models_map.items()))
```

Comment:
why transform it back into a dict when you stop using the dict portion of it? Why not keep it as an iterator?

Reply:
Done